contrib/sbtc/key-admin/contracts/key-admin.clar

```diff
@@ -0,0 +1,68 @@
+
+;; title: key-admin
+;; version:
+;; summary:
+;; description:
+
+;; traits
+;;
+;;(impl-trait tokens-ft)
+
+;; token definitions
+;; 
+(define-fungible-token tokens)
+
+;; constants
+;;
+(define-constant contract-owner tx-sender)
+(define-constant err-invalid-caller (err u1))
+
+;; data vars
+;;
+(define-data-var coordinator (optional principal) none)
+
+;; data maps
+;;
+(define-map signers uint principal)
+
+;; public functions
+;;
+(define-public (set-coordinator-key (key principal))
+    (if (is-valid-caller)
+        (ok (var-set coordinator (some key)))
+        err-invalid-caller
+    )
+)
+
+(define-public (set-signer-key (id uint) (key principal))
+    (if (is-valid-caller)
+        (ok (map-set signers id key))
+        err-invalid-caller
+    )
+)
+
+(define-public (mint! (amount uint))
+    (if (is-valid-caller)
```

Comment:
Instead of gating this method with `is-valid-caller`, could `mint!` accept a frost signature which the method verifies using `secp256k1-verify`? Same with burn! (which needs to be implemented as well).

Reply:
We discussed this in the meeting, I will try to do it in clarity but seems unlikely for short term if ever.